x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiBasicTable, EuiBasicTableColumn, EuiButtonIcon, EuiCopy } from '@elastic/eui';
+import { CriteriaWithPagination } from '@elastic/eui/src/components/basic_table/basic_table';
+import { useActions, useValues } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { CredentialsLogic } from '../credentials_logic';
+import { IApiToken } from '../types';
+import { TOKEN_TYPE_DISPLAY_NAMES } from '../constants';
+import { apiTokenSort } from '../utils/api_token_sort';
+import { getModeDisplayText, getEnginesDisplayText } from '../utils';
+
+export const CredentialsList: React.FC = () => {
+  const { deleteApiKey, fetchCredentials, showCredentialsForm } = useActions(CredentialsLogic);
+
+  const { apiTokens, meta } = useValues(CredentialsLogic);
+
+  const items = useMemo(() => apiTokens.slice().sort(apiTokenSort), [apiTokens]);
+
+  const columns: Array<EuiBasicTableColumn<IApiToken>> = [
+    {
+      name: 'Name',
+      width: '12%',
+      render: (item: IApiToken) => item.name,
+    },
+    {
+      name: 'Type',
+      width: '15%',
+      render: (item: IApiToken) => TOKEN_TYPE_DISPLAY_NAMES[item.type],
+    },
+    {
+      name: 'Key',
+      width: '36%',
+      render: (item: IApiToken) => (
+        <EuiCopy
+          textToCopy={item.key || ''}
+          afterMessage={i18n.translate('xpack.enterpriseSearch.appSearch.credentials.copied', {
+            defaultMessage: 'Copied',
+          })}
+        >
+          {(copy) => (
+            <div>
+              <EuiButtonIcon
+                onClick={copy}
+                iconType="copyClipboard"
+                aria-label={i18n.translate(
+                  'xpack.enterpriseSearch.appSearch.credentials.copyApiKey',
+                  {
+                    defaultMessage: 'Copy API Key to clipboard',
+                  }
+                )}
+              />
+              {item.key}
+            </div>
```

Comment:
nit: one more set of div wrappers we can yeet out

```suggestion
            <>
              <EuiButtonIcon
                onClick={copy}
                iconType="copyClipboard"
                aria-label={i18n.translate(
                  'xpack.enterpriseSearch.appSearch.credentials.copyApiKey',
                  {
                    defaultMessage: 'Copy API Key to clipboard',
                  }
                )}
              />
              {item.key}
            </>
```

Also because I'm sure everyone's wondering, a generic div wrapper did not kill my parents. I them just because 😂

Reply:
lolz d2e21d7407d6477e27304ee1903332e5179d534a